pkg/scheduler/core/generic_scheduler.go

```diff
@@ -229,31 +224,131 @@ func (g *genericScheduler) Preempt(pod *v1.Pod, nodeLister algorithm.NodeLister,
 	if err != nil {
 		return nil, nil, nil, err
 	}
-	nodeToVictims, err := selectNodesForPreemption(pod, g.cachedNodeInfoMap, potentialNodes, g.predicates, g.predicateMetaProducer, g.schedulingQueue, pdbs)
+	nodeToVictims, err := selectNodesForPreemption(pod, g.cachedNodeInfoMap, potentialNodes, g.predicates,
+		g.predicateMetaProducer, g.schedulingQueue, pdbs)
 	if err != nil {
 		return nil, nil, nil, err
 	}
+
+	candidateNodeName, err := g.processPreemptionWithExtenders(pod, nodeToVictims)
+	if len(candidateNodeName) == 0 || err != nil {
+		return nil, nil, nil, err
+	}
+
+	// Lower priority pods nominated to run on this node, may no longer fit on
+	// this node. So, we should remove their nomination. Removing their
+	// nomination updates these pods and moves them to the active queue. It
+	// lets scheduler find another place for them.
+	nominatedPods := g.getLowerPriorityNominatedPods(pod, candidateNodeName)
+	if nodeInfo, ok := g.cachedNodeInfoMap[candidateNodeName]; ok {
+		return nodeInfo.Node(), nodeToVictims[candidateNodeName].Pods, nominatedPods, err
+	} else {
+		return nil, nil, nil, fmt.Errorf(
+			"preemption failed: the target node %s has been deleted from scheduler cache",
+			candidateNodeName)
+	}
+}
+
+// processPreemptionWithExtenders processes preemption with extenders
+func (g *genericScheduler) processPreemptionWithExtenders(
+	pod *v1.Pod,
+	nodeToVictims map[string]*schedulerapi.Victims,
+) (string, error) {
+	// The final candidate node for preemption.
+	var candidateNodeName string
+
+	// Scheduler extenders that do not support preemption
+	nonPreemptingExtenders := []algorithm.SchedulerExtender{}
+
+	if len(nodeToVictims) > 0 {
+		for _, extender := range g.extenders {
+			if extender.SupportsPreemption() {
+				var err error
+				// Replace nodeToVictims with result after preemption from extender.
+				if nodeToVictims, err = extender.ProcessPreemption(pod, nodeToVictims); err != nil {
+					return "", err
+				}
+				// If node list has all removed, no preemption will happen, skip other extenders.
+				if len(nodeToVictims) == 0 {
+					break
+				}
+			} else {
+				nonPreemptingExtenders = append(nonPreemptingExtenders, extender)
+			}
+		}
+	}
+
+	// Unconditional call pickOneNodePassingNonPreemptingExtenders() since if nonPreemptingExtenders is empty,
+	// this will return result of pickOneNodeForPreemption() which is expected.
+	candidateNodeName = g.pickOneNodePassingNonPreemptingExtenders(pod, nodeToVictims, nonPreemptingExtenders)
```

Comment:
As discussed we no longer need to check this with extenders that do not support preemption.

Reply:
Sorry Bobby, the change in `generic_scheduler.go` has not been pushed to remote yet since local test keep failing (probably caused by recent integration test flaky, let me do some rebase and retest).